runtime/sema/checker.go

```diff
@@ -365,11 +381,29 @@ func (checker *Checker) IsChecked() bool {
 	return checker.isChecked
 }
 
+type stopChecking struct{}
+
 func (checker *Checker) Check() error {
 	if !checker.IsChecked() {
 		checker.Elaboration.setIsChecking(true)
 		checker.errors = nil
 		check := func() {
+			if checker.errorShortCircuitingEnabled {
+				defer func() {
+					switch recovered := recover().(type) {
+					case stopChecking:
+						// checking should stop
```

Comment:
If the error is in an import, would it (need to?) terminate checking the original program?

Reply:
That's a good question. I'd say the stop should propagate: If a program is imported, and check it reports an error, then checking the importing program should also stop. 

I'll add a test case for this

Reply:
Added a test case in 57906f4aee6331bfc056d64e379ef3ba553a7982